### DISCUSSION TITLE
Code question: about the git(.*) packages

### DISCUSSION THREAD

**Discussion by dduportal** [updatecli/updatecli#471] [General]
Hello @olblak , while working on #457 , I was wondering why is the git logic split between the `git` generic package and the `git` package itself?

Most of the implementation logic for the methods of `SCMHandler` in https://github.com/updatecli/updatecli/blob/main/pkg/plugins/git/scm.go are using git generic's logic.

The reason is that with #465 incoming, we might want to simplify into a single package, with mocking capability. WDYT?

**Comment by olblak** [comment]
The purpose of the package generic was to factorize its logic so we could distinguish the git package used as a plugin like `gittag`, `scm`, and the git package used by other "git like" plugins such as github scm. This is why the function that implements `git push` is in the generic package while scm interface is not.

I was initially thinking to move the git "generic" under the path "[core](https://github.com/updatecli/updatecli/tree/main/pkg/core)" and keeping everything else under [plugins](https://github.com/updatecli/updatecli/tree/main/pkg/core) but it doesn't make sense anymore.
